src/XR/features/WebXRNearInteraction.ts

```diff
@@ -317,27 +431,23 @@ export class WebXRNearInteraction extends WebXRAbstractFeature {
                         if (indexTipPose && indexTipPose.transform) {
-                            let axisRHSMultiplier = this._scene.useRightHandedSystem ? 1 : -1;
-                            const indexTipPos = indexTipPose.transform.position;
-                            const indexTipOrientation = indexTipPose.transform.orientation;
-                            this._indexTipQuaternion.set(
-                                indexTipOrientation.x,
-                                indexTipOrientation.y,
-                                indexTipOrientation.z * axisRHSMultiplier,
-                                indexTipOrientation.w * axisRHSMultiplier
-                            );
-
-                            // set positions for near pick and hover
-                            if (controllerData.pickIndexMeshTip) {
-                                controllerData.pickIndexMeshTip.position.set(indexTipPos.x, indexTipPos.y, indexTipPos.z * axisRHSMultiplier);
-                            }
-
-                            // set near interaction grab ray parameters
-                            const nearGrabRayLength = this._nearGrabLengthScale * this._hoverRadius;
-                            controllerData.grabRay.origin.set(indexTipPos.x, indexTipPos.y, indexTipPos.z * axisRHSMultiplier);
-                            this._indexTipQuaternion.toEulerAnglesToRef(this._indexTipOrientationVector);
-                            controllerData.grabRay.direction.set(this._indexTipOrientationVector.x, this._indexTipOrientationVector.y, this._indexTipOrientationVector.z);
-                            controllerData.grabRay.length = nearGrabRayLength;
+                            this._processTouchPoint(id, indexTipPose.transform);
+                        }
+                    }
+                }
+                else if (controllerData.xrController.inputSource.gamepad) {
+                    const controller = controllerData.xrController.inputSource;
+                    if (this._options.nearInteractionControllerMode === WebXRNearControllerMode.CENTERED_ON_CONTROLLER) {
+                        // Try to get the grip space, with a target ray space fallback
+                        let controllerPose = _xrFrame.getPose(controller.gripSpace || controller.targetRaySpace, this._xrSessionManager.referenceSpace);
+                        if (controllerPose && controllerPose.transform) {
+                            this._processTouchPoint(id, controllerPose.transform);
+                        }
+                    }
+                    else if (this._options.nearInteractionControllerMode === WebXRNearControllerMode.CENTERED_IN_FRONT) {
+                        let controllerPose = _xrFrame.getPose(controller.targetRaySpace, this._xrSessionManager.referenceSpace);
+                        if (controllerPose && controllerPose.transform) {
+                            this._processTouchPoint(id, controllerPose.transform);

```

Comment:
The only difference here is the xr space used, you could save a few lines removing the repeated code. Unless i totally missed something :-)